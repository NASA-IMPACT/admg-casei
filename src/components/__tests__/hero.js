import React from "react"
import renderer from "react-test-renderer"
import Hero from "../hero"

const testTagTitle = "Instrument"
const testTitle = "2D-C/P"
const testSubTitle = "2D-C/P Hydrometeor Imaging Probe"
const testDescription = "The 2DC probe records images of hydrometeors..."
const testTextToImageRatio = [5, 3]
const testId = "test"

describe("Header", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <Hero
          tagTitle={testTagTitle}
          title={testTitle}
          subTitle={testSubTitle}
          description={testDescription}
          id={testId}
        >
          <div>children</div>
        </Hero>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it("renders with adjust ratios", () => {
    const tree = renderer
      .create(
        <Hero
          tagTitle={testTagTitle}
          title={testTitle}
          subTitle={testSubTitle}
          description={testDescription}
          textToImageRatio={testTextToImageRatio}
        >
          <div>children</div>
        </Hero>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
