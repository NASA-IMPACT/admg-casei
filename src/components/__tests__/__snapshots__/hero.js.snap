// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<section
  data-cy="test-hero"
  style={
    Object {
      "alignContent": "center",
      "display": "grid",
      "gridTemplateColumns": "1fr minmax(auto,  1280px) 1fr",
      "marginBottom": "6rem",
      "minHeight": "35rem",
      "width": "100vw",
    }
  }
>
  <div
    style={
      Object {
        "display": "grid",
        "gridArea": "1 / 2 / 1 / 2",
        "gridGap": ".5rem",
        "gridTemplateColumns": "3fr 5fr",
        "padding": "0 5rem",
      }
    }
  >
    <div
      style={
        Object {
          "alignSelf": "end",
          "gridArea": "1 / 1 / 1 / 1",
        }
      }
    >
      <p
        style={
          Object {
            "textTransform": "uppercase",
          }
        }
      >
        Instrument
      </p>
      <h1>
        2D-C/P
      </h1>
      <p
        style={
          Object {
            "fontSize": "x-large",
          }
        }
      >
        2D-C/P Hydrometeor Imaging Probe
      </p>
    </div>
    <div
      style={
        Object {
          "alignSelf": "start",
          "gridArea": "2 / 1 / 2 / 1",
        }
      }
    >
      <p>
        The 2DC probe records images of hydrometeors...
      </p>
    </div>
    <div
      style={
        Object {
          "gridArea": "1 / 2 / 3 / 3",
        }
      }
    >
      <div>
        children
      </div>
    </div>
  </div>
</section>
`;

exports[`Header renders with adjust ratios 1`] = `
<section
  data-cy="undefined-hero"
  style={
    Object {
      "alignContent": "center",
      "display": "grid",
      "gridTemplateColumns": "1fr minmax(auto,  1280px) 1fr",
      "marginBottom": "6rem",
      "minHeight": "35rem",
      "width": "100vw",
    }
  }
>
  <div
    style={
      Object {
        "display": "grid",
        "gridArea": "1 / 2 / 1 / 2",
        "gridGap": ".5rem",
        "gridTemplateColumns": "5fr 3fr",
        "padding": "0 5rem",
      }
    }
  >
    <div
      style={
        Object {
          "alignSelf": "end",
          "gridArea": "1 / 1 / 1 / 1",
        }
      }
    >
      <p
        style={
          Object {
            "textTransform": "uppercase",
          }
        }
      >
        Instrument
      </p>
      <h1>
        2D-C/P
      </h1>
      <p
        style={
          Object {
            "fontSize": "x-large",
          }
        }
      >
        2D-C/P Hydrometeor Imaging Probe
      </p>
    </div>
    <div
      style={
        Object {
          "alignSelf": "start",
          "gridArea": "2 / 1 / 2 / 1",
        }
      }
    >
      <p>
        The 2DC probe records images of hydrometeors...
      </p>
    </div>
    <div
      style={
        Object {
          "gridArea": "1 / 2 / 3 / 3",
        }
      }
    >
      <div>
        children
      </div>
    </div>
  </div>
</section>
`;
